### PR TITLE
Wrapping up various p1 Zupoll tasks

### LINKED ISSUE #48: Polls are retrieved in a different order from their posting
This means that questions that are supposed to be in sequence don't display as such.

### LINKED ISSUE #50: Be able to view the votes you made on a poll
Right now, we don't store the options users actually voted on. I'm not sure why this wasn't a priority earlier, but it means that the server & client has no way of linking votes.

However, I think that guarantee is (1) not even appreciated/understood by people using zupoll and (2) not worth the UX headache of not knowing what you voted for.

### LINKED ISSUE #51: Enforce that there is only vote per poll in a MultiVoteRequest
Caught by Sampriti

### PR DESCRIPTION
**Pull request by vb7401**
Resolves #50. Stores user's votes in localStorage and reads from there when highlighting options.

Resolves #51. Checks that there are not multiple votes per poll ID.

Resolves #48. Changing the database prisma schema would require a reset of the data, as I wasn't careful with migrating the schema step by step. After trying to deal with that for a while, I decided on a jank but functional solution that didn't change the schema. On creation, the client puts the order of the polls in the ID field, and then the server stores this order in the options array. When the polls are retrieved by the client, it automatically sorts the polls and removes the extra option. It's obviously imperfect and ugly but it gets the job done.

### CODE REVIEW

**Review by ichub — APPROVED** [review]
Make sure you've tested all of this! I Can think of three levels of testing:

1. create a bunch of polls locally 
2. import polls from production into your local database (idk what sql software you're using but mine (DBeaver) lets me export a table as `insert` statements, which I then run in my localdb.
3. run in staging

I think when changes include updates to how we read/write from the database, and also some fancy client-side logic, it can be helpful to include your testing strategy in your PR description. It's not always strictly necessary, but it's easier to understand than reading through all the code, for example.

You can see an example for how I did this in the description + follow-up comments in this PR: https://github.com/proofcarryingdata/zupass/pull/245

tl;dr I think code-wise this is close enough that I don't want to leave a bunch of nitpicky comments, the only thing I want to be certain of is that this works the way you expect it to and you thought through all the edge cases.

**Comment by vb7401** [comment]
Tested on staging with advisory votes and straw polls. The results you voted for do show up, and the ordering of questions has been correct with no extraneous poll options for polls that came before the ordering code was added.
